src/hook/useAppBackgroundApiReconnect.ts

```diff
@@ -0,0 +1,27 @@
+import {useEffect} from 'react';
+import {useApiReconnect} from 'context/ChainApiContext';
+import {useNavigation} from '@react-navigation/core';
+import {apiLoadingScreen} from 'src/navigation/routeKeys';
+import {useAppState} from './useAppState';
+
+export function useAppBackgroundApiReconnect() {
+  const {canReconnect, reconnect} = useApiReconnect();
+  const navigation = useNavigation();
+  const {didAppCameToForeground, setDidAppCameToForeground} = useAppState();
+
+  useEffect(() => {
+    if (didAppCameToForeground && canReconnect) {
+      /**
+       * make it false once used, otherwise it will try to reconnect on any
+       * type of disconnect event. Such as we don't want automatic reconnect
+       * when the app is already in foreground.
+       */
+      setDidAppCameToForeground(false);
```

Comment:
This should work but seems to me it would be easier if we remove the `useAppState` hook and put the App state listener here. because the other hook wouldn't be reusable in any other place when we change it's state here.

Reply:
Bringing in the AppState listener does not solve the problem. [Here](https://github.com/litentry/litentry-app/pull/592/commits/6996d7066451a66411088fd13cfbde78d6d6b122) is a different implementation.